static/js/message_list_data.js

```diff
@@ -397,7 +397,7 @@ export class MessageListData {
         } else {
             // Any of the ids that we skipped over (due to them being local-only) might be the
             // closest ID to the desired one, in case there is no exact match.
-            potential_closest_matches.unshift(_.last(potential_closest_matches) - 1);
+            potential_closest_matches.unshift(closest - 1);
```

Comment:
Why is this replacement correct?

Reply:
Oh, I guess we're pushing items onto the end, not the front; the `closest -= 1` logic in the loop means that perhaps this should be `closest`, not `closest - 1`?

Reply:
We just unconditionally pushed `closest` onto the end of the array 7 lines ago, and we did not execute the `if` branch 4 lines ago because we’re in the `else` branch, so the last element of the array still equals `closest`. The `- 1` is preserved from the old code.

I haven’t investigated whether this behavior is correct beyond showing that it’s the same.

Reply:
Oh, I missed the extra push outside the loop.  OK.